### PR TITLE
chore: upgrade react dependencies set 1 [DET-3649]

### PR DESCRIPTION
**Pull request by hamidzr**
## Description

excluded `history` and `antd` from this upgrade as they had a few breaking changes that needed more work.

Notable upgrades:
@typescript-eslint 2.34 => 3.6: https://github.com/typescript-eslint/typescript-eslint/releases

## Commentary (optional)

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234

### CODE REVIEW

File: webui/react/config-overrides.js [hunk 1]
```diff
@@ -23,7 +23,7 @@ module.exports = override(
   }),
 
   // Add LESS loader support for antd.
-  addLessLoader({ javascriptEnabled: true }),
+  addLessLoader({ lessOptions: { javascriptEnabled: true } }),
```
Comment (hkang1) [inline webui/react/config-overrides.js:26]: **question:** did I just set this wrong or was this a change due to an upgraded `customize-cra` version?

File: webui/react/src/contexts/index.ts [hunk 2]
```diff
@@ -50,7 +50,11 @@ export const generateContext = <T, A = Action<T>>(options: Options<T, A>): Expor
   };
   const reducer = options.reducer || defaultReducer;
 
-  const Provider: React.FC = (props: PropsWithChildren<{}>) => {
+  interface Props {
+    children?: React.ReactNode
+  }
+
+  const Provider: React.FC<Props> = (props: Props) => {
```
Comment (hkang1) [inline webui/react/src/contexts/index.ts:57]: **question:** was there a version change that broke `PropsWithChildren`?

File: webui/react/.eslintrc.js [hunk 3]
```diff
@@ -59,6 +59,7 @@ module.exports = {
       beforeSelfClosing: 'always',
       closingSlash: 'never',
     } ],
+    'react/prop-types': 'off',
```
Comment (hkang1) [inline webui/react/.eslintrc.js:62]: **question:** isn't this a good thing to keep? maybe something was broken after eslint / plugins updated?

**Comment by hkang1** [comment]
I ran `npm run lint` and ran into some warnings.

I think this will at least allow `any` definition in function arguments.
```
    '@typescript-eslint/explicit-module-boundary-types': [ 'error', {
      allowArgumentsExplicitlyTypedAsAny: true,
    } ],
```

File: webui/react/config-overrides.js (hunk 1 above)
Comment (hamidzr) [reply webui/react/config-overrides.js:26]: less loader upgrade now needs less options to be inside an object . one of the breaking changes.

**Comment by hamidzr** [comment]
> LGTM! There is one minor warning that comes up:
> 
> ```
> /Users/calebkang/Projects/determined/webui/react/src/components/Link.tsx
>   41:5  warning  React Hook useCallback received a function whose dependencies are unknown. Pass an inline function instead  react-hooks/exhaustive-deps
> ```

yep, I saw that but didn't want to fix that in this PR that could lead to more changes not necessarily a good reason to delay upgrading everything. and that warning is not a problem this PR is introducing. wdy think

**Comment by hkang1** [comment]
> > LGTM! There is one minor warning that comes up:
> > ```
> > /Users/calebkang/Projects/determined/webui/react/src/components/Link.tsx
> >   41:5  warning  React Hook useCallback received a function whose dependencies are unknown. Pass an inline function instead  react-hooks/exhaustive-deps
> > ```
> 
> yep, I saw that but didn't want to fix that in this PR that could lead to more changes not necessarily a good reason to delay upgrading everything. and the reason for that warning is not introduced by this PR. wdy think

I see, let's leave for a different PR.

**Pull request merged**